src/go/rpk/pkg/schemaregistry/client.go

```diff
@@ -0,0 +1,72 @@
+package schemaregistry
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/net"
+	"github.com/spf13/afero"
+	"github.com/twmb/franz-go/pkg/sr"
+)
+
+func NewClient(fs afero.Fs, p *config.RpkProfile) (*sr.Client, error) {
+	api := &p.SR
+
+	d := p.Defaults()
+	if len(api.Addresses) == 0 && d.NoDefaultCluster {
+		return nil, errors.New("no schema registry hosts specified and rpk.yaml is configured to not use a default cluster")
+	}
+
+	urls := p.SR.Addresses
+	for i, u := range p.SR.Addresses {
+		scheme, _, err := net.ParseHostMaybeScheme(u)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse your schema registry address %q: %v", u, err)
+		}
+		switch scheme {
+		case "http", "https":
+			continue
+		case "":
+			if p.SR.TLS != nil {
+				urls[i] = "https://" + u
+			} else {
+				urls[i] = "http://" + u
+			}
+		default:
+			return nil, fmt.Errorf("unsupported scheme %q in the schema registry address %q", scheme, u)
+		}
```

Comment:
I'm looking at this again and I think this should go in the params loading code, when initializing the virtual profile?
OK to live here for now but, maybe worth moving in follow up